### PR TITLE
Update Whisper to use fetch helper

### PR DESCRIPTION
**Pull request by francislata**
## Overview
This is just a breakdown of #2366 to smaller PRs. This one starts with Whisper.

### CODE REVIEW

**Comment by geohot** [comment]
Why is it still called "Cache model weights"? Isn't it caching all the downloads?

**Comment by francislata** [comment]
> Why is it still called "Cache model weights"? Isn't it caching all the downloads?

@geohot - let me update that to "Cache downloads" to make it more generic.

**Comment by francislata** [comment]
@geohot - all updated.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -81,13 +81,11 @@ jobs:
...
-    - name: Cache model weights
+    - name: Cache downloads
       uses: actions/cache@v3
       with:
-        path: |
-          weights/whisper-tiny.en.pt
-          weights/gpt2.tiktoken
-        key: model-weights-v1
+        path: ~/.cache/tinygrad/downloads/
+        key: model-weights
```
Comment (geohot) [inline .github/workflows/test.yml:88]: Please read the diff and ask what wouldn't make sense to someone reading it.

**Pull request merged**